From 1055874dd925d2d75037e4125aaa1294c2734a73 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Thu, 10 Sep 2015 16:32:28 +0800
Subject: [PATCH] MLK-11501 rtc: rtc-snvs: fix the kernel dump with poweroff
 test

The function .snvs_poweroff() get the un-corrected rtc snvs clock that
causes kernel dump. It was introduced by the commit: b2e6abde4321.

Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 drivers/rtc/rtc-snvs.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/rtc/rtc-snvs.c b/drivers/rtc/rtc-snvs.c
index 880edfc..cb19c83 100644
--- a/drivers/rtc/rtc-snvs.c
+++ b/drivers/rtc/rtc-snvs.c
@@ -44,6 +44,7 @@ struct snvs_rtc_data {
 };
 
 static void __iomem *snvs_base;
+static struct clk *clk_snvs;
 
 static u32 rtc_read_lp_counter(void __iomem *ioaddr)
 {
@@ -261,14 +262,12 @@ static irqreturn_t snvs_rtc_irq_handler(int irq, void *dev_id)
 static void snvs_poweroff(void)
 {
 	u32 value;
-	struct snvs_rtc_data *data = container_of(snvs_base, struct snvs_rtc_data,
-						  ioaddr);
 
-	clk_enable(data->clk);
+	clk_enable(clk_snvs);
 	value = readl(snvs_base + SNVS_LPCR);
 	/* set TOP and DP_EN bit */
 	writel(value | 0x60, snvs_base + SNVS_LPCR);
-	clk_disable(data->clk);
+	clk_disable(clk_snvs);
 }
 
 static int snvs_rtc_probe(struct platform_device *pdev)
@@ -291,6 +290,7 @@ static int snvs_rtc_probe(struct platform_device *pdev)
 		dev_err(&pdev->dev, "can't get snvs-rtc clock\n");
 		data->clk = NULL;
 	}
+	clk_snvs = data->clk;
 
 	data->irq = platform_get_irq(pdev, 0);
 	if (data->irq < 0)
-- 
1.8.0

